lib/common.js

```diff
@@ -338,13 +338,26 @@ function ensureEntryExists(req, res, next) {
 
 /*
  * This function is used to abstract away the logic involved in checking
- * that a particular account uuid has snaplinks enabled. Today, it is
- * called in the `ensureSnaplinksEnabled` method defined in this module
- * and in the putlinkHandler chain to verify that the source object (the
- * object being linked to) is not owned by an account for which snaplinks
- * are enabled.
+ * that a particular account uuid potentially has existing SnapLinks.
+ * An account is assumed to have no SnapLinks if:
+ *
+ *  * they are in the accountsSnaplinksDisabled array; or
+ *  * we have completed the audit and "fixed" the system so that there are no
+ *    longer *any* SnapLinks for any account. If we upgraded from Manta v1, the
+ *    upgrade will have set "SNAPLINK_CLEANUP_REQUIRED". When that is gone, the
+ *    cleanup has been completed.
+ *
+ * in either case, next() will be called with `false`. Otherwise next() will be
+ * called with `true`.
  */
-function checkAccountSnaplinksEnabled(req, uuid, next) {
+function checkAccountSnaplinksMightExist(req, uuid, next) {
+    if (req.config.snaplinkCleanupRequired !== true) {
+        // We've run the cleanup and this Manta is SnapLink-free, so snaplinks
+        // will not exist for this account.
```

Comment:
Or the comment could acknowledge this may have been a mantav2 that only started after snaplinks were removed as a feature, so cleanup was never necessary.

Reply:
ok